Libraries/Components/RefreshControl/RefreshControl.js

```diff
@@ -137,11 +153,11 @@ const RefreshControl = React.createClass({
   },
 
   _onRefresh() {
+    this._lastNativeRefreshing = true;
+
     this.props.onRefresh && this.props.onRefresh();
 
-    if (this._nativeRef) {
-      this._nativeRef.setNativeProps({refreshing: this.props.refreshing});
-    }
+    this.forceUpdate();
```

Comment:
We need to run this code after the next render cycle since the user should call `setState` in the `onRefresh` function we cannot just call setNativeProps since there is no guarentee that the state will be updated synchronously.

This also makes sure that if the user doesn't update the state in the `onRefresh` function the component will still update and make the refresh control stop refreshing immediatly.